### PR TITLE
Use opamconfig in flac.0.1.2

### PR DESCRIPTION
**Pull request by michipili**

### CODE REVIEW

File: packages/flac/flac.0.1.2/opam [hunk 1]
```diff
@@ -3,15 +3,18 @@ maintainer: "Romain Beauxis <toots@rastageeks.org>"
...
+  ["./configure" "--prefix" prefix opamconfig:cflags opamconfig:ldflags opamconfig:ocamlflags]
   [make]
 ]
 install: [
   [make "install"]
 ]
 remove: ["ocamlfind" "remove" "flac"]
-depends: ["ocamlfind" "ogg"]
+depends: [
+  "opamconfig"
```
Comment (yallop) [inline packages/flac/flac.0.1.2/opam]: Should `opamconfig` be a `{build}` dependency (here and in the other PRs)?

Comment (lefessan) [inline packages/flac/flac.0.1.2/opam]: Yes, it should, I think. The code does not depend on what is detected by `opamconfig`, so there is no reason to recompile when `opamconfig` is updated.

**Comment by yallop** [comment]
Changing all these packages, which are currently working, is going to create additional work for their maintainers, add additional dependencies, and potentially introduce bugs.  I think it'd be better to advertise `opamconfig` and let people adopt it for their own packages as they see fit.

**Comment by dsheets** [comment]
I agree with @yallop. It's not clear what means `opamconfig` uses to divine its flags nor what the packages in these PRs gain from this change. In some cases, using `opamconfig` 0.1 may not work or be appropriate (e.g. #6780). Finally, it's best to notify the maintainers of the packages that you are proposing changes to especially when you are adding a new, opaque dependency to their packages where before there was simply a `configure` command line.

**Comment by michipili** [comment]
> I agree with @yallop. It's not clear what means opamconfig uses to divine its flags nor what the packages in these PRs gain from this change

The point is that the installation instructions **are** already broken. You just happen to not see it. In most cases, the packages I changed *do not install* on my system.  Deducing configure flags from the *os* variable is the problem. Worse, there is no possibility to override the flags passed from the broken package definition.

I understand that my solution is not perfect, but the problem is here and needs to be solved.

**Comment by michipili** [comment]
@avsm Thank you. It is probably easier to close all these PRs right now, and move the full discussion on #6271. What do you think?

**Comment by dsheets** [comment]
Deducing `configure` flags from the `os` variable and having the ability to set them per-package seems preferable to guessing based on the totally unrelated `PATH` environment variable. In the case of `ssl`, there is explicitly a way to override the flag whereas with the current version of `opamconfig` there is no way to override them.

In general, I agree with you that using *only* `os` is not complete but is, currently without programmatic detection, the best predictor of the correct flags to use. I also agree that there is a problem. Where we disagree is on the form of the solution (a shell script?) and the modification of packages (to add a new dependency!) without any of

1. issue reports upstream
2. issue reports to this repository
3. notification of the maintainer

There is also the issue that the `opam` build instructions are now much less clear than they were before. No longer can the user see that new `-I` includes are being added to `CPPFLAGS` instead only understanding that some magical detection process has selected *all* of the `CPPFLAGS` to use. This needs to be made much more clear.

@avsm if one is concerned with making packages work on FreeBSD and OpenBSD, two platforms notorious for their consistency, I think that the best way to do this is to contribute new build invocations that work for these platforms *and* make sure that the upstream packages adopt them so that new versions keep working on those operating systems. Combined with some of the above suggestions, I think this will really improve the quality of the binding packages in the repository but as it stands, I don't think we can merge these PRs.

**Comment by avsm** [comment]
Yes, let's move to #6721 so we have one place to discuss it. thanks!

**Pull request closed without merging**